waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -141,9 +146,18 @@ isDbRunning ::
 isDbRunning genProjectDir = do
   chan <- newChan
   exitCode <- DbJobs.dbExecuteTest genProjectDir chan
-  -- NOTE: We only care if the command succeeds or fails, so we don't look at
-  -- the exit code or stdout/stderr for the process.
-  return $ exitCode == ExitSuccess
+
+  case exitCode of
+    ExitSuccess -> return True
+    ExitFailure _ -> do
+      textOutput <- collectJobTextOutputUntilExitReceived chan
+
+      -- "Database not created" error is fine since Prisma will create it for us.
+      let isErrorTolerated = any containsDatabaseNotCreatedError textOutput
+      return isErrorTolerated
+  where
+    -- Prisma error code for "Database not created" is P1003.
+    containsDatabaseNotCreatedError = T.isInfixOf "P1003"
```

Comment:
So, while this will work in 99% cases, what if the prisma error text contains `PIP1003` or `P10030` or anything else that is not `\bP1003\b`? Then it will give false positive.

So to be 100% safe, I would do a bit fancier check here. Probably use regex `\bP1003\b`.